Increase space next to checkmark on selectable items

ryantmer opened the pull request:
Just an idea, mostly figured I'd do a quick mock and then look for feedback.

This makes it more obvious that these are selectable/checkable items. If this something we want, might want to consider adding the traditional circle next to d2l-menu-item-radio as well.

![image](https://cloud.githubusercontent.com/assets/6231623/19402138/5bd60ec2-922d-11e6-85d0-e14a87bac0f5.png)

dbatiste commented:
Yeah I was wondering about this too.  Another thing I was wondering about is how close the image is to the text.  Seems a little cramped.

ryantmer commented:
Yeah, it kinda does with the border especially. My best suggestion (as a not-designer) would be to have the gap on either side of the checkbox/box equal - so add a [1rem space](https://github.com/Brightspace/d2l-menu-ui/blob/master/d2l-menu-item.html#L15) between box and text?

ryantmer commented:
Er...

![image](https://cloud.githubusercontent.com/assets/6231623/19402509/765e70f2-922f-11e6-93fd-770118c8228a.png)

CSS confuses me sometimes - the padding is 1rem, but adding a `margin-right: 1rem` to the checkbox border div is apparently a larger space...

dbatiste commented:
I think I would try following a similar approach as [d2l-menu-item-return](https://github.com/Brightspace/d2l-menu-ui/blob/master/d2l-menu-item-return.html).  It appears to be `1rem` (20px).

ryantmer commented:
So I realized that `d2l-menu-item-return` is using flexbox, which makes a lot of sense, so I updated both checkbox and radio to use that. Right now the radio items just have `border-color: 1px transparent`, but that could change to be a circle or something, perhaps.

![image](https://cloud.githubusercontent.com/assets/6231623/19442775/81a1a346-9458-11e6-8f13-c4046b35f260.png)

![image](https://cloud.githubusercontent.com/assets/6231623/19442829/b71c65c4-9458-11e6-9bd5-d4212bf9e8f3.png)

I _think_ there is still about 1px more on the right, but it is because of the text/font (since padding of item == margin-left of border box == 1rem now)... not sure if there's a way to prevent that.

dbatiste commented:
Sounds good.  Yeah, something else to note: the menu items are supposed to get truncated with ellipsis when too long.  The original menu-items handle that, and so if checkbox and radio items implement the same styles or follow the same pattern, then it will be consistent.

Is it the 1px border that is resulting in the extra pixel?  If so, it we may be able to calc the margin with something like `calc(1rem - 1px)`, if we care that much.  Alternatively margin could be set to 0.95rem instead - the difference at different font-sizes would not be super significant, and all the items in the menu ought to be the same type so it would not be obvious if it was off by 1px at those alternate font-sizes.

ryantmer commented:
Yeah, noticed that was missing so I implemented it in 9f713af (it's within `d2l-menu-item` itself, not the `d2l-menu-item-styles` - not sure if it should be moved?). Looks to be working as expected, but I've updated the demos to explicitly exercise that.

As for the extra pixel, it could be the extra pixel, although setting `box-sizing: border-box` and upping the width/height of the icon border gave the same result. In the end, I went with just shrinking the margin to 0.9rem, as that appears to make it more-or-less equally spaced from the left side and the text.

geurts commented on d2l-menu-item-selectable-styles.html:
```diff
@@ -6,25 +6,55 @@
 				visibility: visible;
 			}
+
+			.icon-border {
```
Nice job with the changes, i was looking at doing this myself (the alignment and padding part).

There's actually an icon for both the unchecked box and the checked box. Can we use those instead? Here are two links:

- [Unchecked](https://camo.githubusercontent.com/b814decf4bd8fea54d3d779a1b1b9b26d26de75a/68747470733a2f2f63646e2e7261776769742e636f6d2f427269676874737061636555492f69636f6e732f6d61737465722f696d616765732f74696572322f636865636b2d626f782d756e636865636b65642e7376673f7261773d74727565)
- [Checked](https://camo.githubusercontent.com/b8faa4e128728b04b97900b5bd7f5128fba3c8c9/68747470733a2f2f63646e2e7261776769742e636f6d2f427269676874737061636555492f69636f6e732f6d61737465722f696d616765732f74696572322f636865636b2d626f782e7376673f7261773d74727565)

ryantmer commented:
After some discussion with Nicole and Bob, looks like we're going to drop the empty box for the checkbox items - just a blank space, and a (centred) checkmark when the item is selected. There design for My Courses filtering, however, requires the box... I'll look into what the best way to implement that is, but for now in this PR I will remove the box.

ryantmer commented:
Removed the border at least for now, giving the results shown below (meaning the effect of this PR is now really "give more space to the checkmark"). Seem okay?

![image](https://cloud.githubusercontent.com/assets/6231623/19705610/6e6b0982-9adb-11e6-9c69-4f9f00e91989.png)

dbatiste commented:
I am curious why the design is not consistent... ?  Is the purpose/meaning the same?

ryantmer commented:
Discussion notes, for completeness:

- In the context of `d2l-menu`, single-select items or multi-select items are still, at their core, _menu_ components, and should act as such.
- The checkboxes that appear in the My Courses overlay for filtering by semester/department are form-like, so it doesn't necessarily make sense for them to be here. (I'll likely have some follow-up conversations about this, as while they are form-like, they're not actually a form... but I'll figure that out.)
- For now at least, `d2l-my-courses-ui` will continue to use custom components for its checkboxes, to not muddy the semantics of `d2l-menu`. It can be updated if required.

dbatiste commented on demo/menu-checkbox.html:
```diff
@@ -14,7 +14,7 @@
 
 			<d2l-menu id="menu" label="Checkbox Menu" style="max-width:400px;">
-				<d2l-menu-item-checkbox text="Checkbox 1" value="1"></d2l-menu-item-checkbox>
+				<d2l-menu-item-checkbox text="Checkbox 1 with a very long title that will be cut off" value="1"></d2l-menu-item-checkbox>
```
Minor nitpick - value not unique.
